internal/idp_test/util_test.go

```diff
@@ -0,0 +1,97 @@
+package idp_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"reflect"
+)
+
+// deepEqual is the same as reflect.DeepEqual, but taking "null" attributes into account.
+func deepEqual(a, b interface{}) bool {
+	if a == nil || b == nil {
+		return equalsNil(a) && equalsNil(b)
+	}
+	v1 := reflect.ValueOf(a)
+	v2 := reflect.ValueOf(a)
+	if v1.Type() != v2.Type() {
+		return false
+	}
+	switch a := a.(type) {
+	case []interface{}:
+		if len(a) == 0 {
+			return equalsNil(b)
+		}
+		b := b.([]interface{})
+		if len(a) != len(b) {
+			return false
+		}
+		for i, v := range a {
+			if !deepEqual(v, b[i]) {
+				return false
+			}
+		}
+		return true
+	case map[string]interface{}:
+		am := nonNilAttributes(a)
+		bm := nonNilAttributes(b.(map[string]interface{}))
+		for k, v := range am {
+			if !deepEqual(v, bm[k]) {
+				return false
+			}
+		}
+		return true
+	default:
+		return reflect.DeepEqual(a, b)
+	}
+}
+
+// equalsNil checks whether the given value is nil in terms of SCIM.
+// e.g. empty array "[]" == null.
+func equalsNil(a interface{}) bool {
+	switch a := a.(type) {
+	case []interface{}:
+		return len(a) == 0
+	case map[string]interface{}:
+		return len(a) == 0
+	default:
+		return a == nil
+	}
+}
+
+// nonNilAttributes removes all nested nil attributes from the given resource.
```

Comment:
Why is this important?

Reply:
SCIM does not differentiate between `null`, `[]` and omitting attributes. This ensures that all these values are omitted, which makes checking if they're equal easier.

Reply:
I see, but should we actually care about that here? Can't we just use `reflect.DeepEqual` and edit the expected responses so that it succeeds?

Reply:
This is possible, but would result in a lot of manual overwrites of values.
Except from remove 'null' values, the tests use `reflect.DeepEqual`.

Reply:
> This is possible, but would result in a lot of manual overwrites of values.

What do you mean by 'manual overwrites'? Are these JSON files currently just copy-pasted? Anyway, SCIM also ignores case differences for property keys, so should we take that into account as well then?

Reply:
> Are these JSON files currently just copy-pasted?

Yes, and I'd prefer to keep it that way.

> Anyway, SCIM also ignores case differences for property keys, so should we take that into account as well then?

Yes, good point. I will fix this.